module/zfs/zpl_file.c

```diff
@@ -216,6 +216,30 @@ zpl_aio_fsync(struct kiocb *kiocb, int datasync)
 #error "Unsupported fops->fsync() implementation"
 #endif
 
+static inline int
+zfs_io_flags(struct kiocb *kiocb)
+{
+	int flags = 0;
+
+#if defined(IOCB_DSYNC)
```

Comment:
Nit - `struct kiocb` was introduced with these `IOCB_` flag constants and both are in the same header scope.  Are the conditionals necessary?

Reply:
I see what you're saying, but I'd suggest we keep them to protect against unexpected kernel changes.

Reply:
my comment was wrong since in earlier kernels the new sync flags didn't exist while the struct did -- so the conditional code is correct.